clients/imodelhub/src/IModelHubBackend.ts

```diff
@@ -7,26 +7,32 @@
  */
 
 import { join } from "path";
-import { ProgressCallback, UserCancelledError } from "@bentley/itwin-client";
-import {
-  AcquireNewBriefcaseIdArg, BackendHubAccess, BriefcaseDbArg, BriefcaseIdArg, BriefcaseLocalValue, BriefcaseManager, ChangesetArg, ChangesetRangeArg, CheckpointArg,
-  CheckpointProps, CreateNewIModelProps, IModelDb, IModelHost, IModelIdArg, IModelJsFs, IModelNameArg, ITwinIdArg, LockMap, LockProps, LockState, SnapshotDb, TokenArg,
-  V2CheckpointAccessProps,
+import type { ProgressCallback} from "@bentley/itwin-client";
+import { UserCancelledError } from "@bentley/itwin-client";
+import type {
+  AcquireNewBriefcaseIdArg, BackendHubAccess, BriefcaseDbArg, BriefcaseIdArg, ChangesetArg, ChangesetRangeArg, CheckpointArg,
+  CheckpointProps, CreateNewIModelProps, IModelIdArg, IModelNameArg, ITwinIdArg, LockMap, LockProps, LockState, TokenArg,
+  V2CheckpointAccessProps} from "@itwin/core-backend";
+import { BriefcaseLocalValue, BriefcaseManager, IModelDb, IModelHost, IModelJsFs, SnapshotDb,
 } from "@itwin/core-backend";
-import { BentleyError, BriefcaseStatus, GuidString, Id64String, IModelHubStatus, IModelStatus, Logger, OpenMode } from "@itwin/core-bentley";
+import type { GuidString, Id64String} from "@itwin/core-bentley";
+import { BentleyError, BriefcaseStatus, IModelHubStatus, IModelStatus, Logger, OpenMode } from "@itwin/core-bentley";
+import type { ChangesetFileProps, ChangesetId, ChangesetIndex, ChangesetIndexAndId, ChangesetProps, CodeProps, IModelVersion, LocalDirName} from "@itwin/core-common";
 import {
-  BriefcaseIdValue, ChangesetFileProps, ChangesetId, ChangesetIndex, ChangesetIndexAndId, ChangesetProps, CodeProps, IModelError, IModelVersion, LocalDirName,
+  BriefcaseIdValue, IModelError,
 } from "@itwin/core-common";
 import { IModelBankClient } from "./imodelbank/IModelBankClient";
-import { IModelClient } from "./IModelClient";
+import type { IModelClient } from "./IModelClient";
 import { BriefcaseQuery } from "./imodelhub/Briefcases";
 import { ChangeSet, ChangeSetQuery, ChangesType } from "./imodelhub/ChangeSets";
 import { CheckpointQuery } from "./imodelhub/Checkpoints";
-import { CheckpointV2, CheckpointV2Query } from "./imodelhub/CheckpointsV2";
+import type { CheckpointV2} from "./imodelhub/CheckpointsV2";
+import { CheckpointV2Query } from "./imodelhub/CheckpointsV2";
 import { IModelHubClient } from "./imodelhub/Client";
 import { CodeQuery } from "./imodelhub/Codes";
 import { IModelQuery } from "./imodelhub/iModels";
-import { Lock, LockLevel, LockQuery, LockType } from "./imodelhub/Locks";
+import type { LockLevel} from "./imodelhub/Locks";
+import { Lock, LockQuery, LockType } from "./imodelhub/Locks";
 import { VersionQuery } from "./imodelhub/Versions";
 import { AzureFileHandler } from "./itwin-client/AzureFileHandler";
 
```

Comment:
I would have preferred to make barrel files list all package exports explicitly and add `type` annotations there rather than sprinkling them all over the source code. Has this option been considered?

Reply:
This would be nice too, but from what ive seen in this massive codebase, we're way too lazy to explicitly do that. 
Most barrels ive seen just blindly export everything from some other file. 

```ts
export * from "./foo/bar";
```

Not sure how much of an effort it'd take to go back in all the barrel files and correctly add the `type` annotation everywhere